modules/oe_theme_helper/src/Plugin/ImageEffect/RetinaScaleImageEffect.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme_helper\Plugin\ImageEffect;
+
+use Drupal\Component\Utility\Image;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Image\ImageInterface;
+use Drupal\image\Plugin\ImageEffect\ScaleImageEffect;
+
+/**
+ * Scales an image resource.
+ *
+ * @ImageEffect(
+ *   id = "retina_image_scale",
+ *   label = @Translation("Retina Scale"),
+ *   description = @Translation("Scaling will maintain the aspect-ratio of the original image. If only a single dimension is specified, the other dimension will be calculated.")
+ * )
+ */
+class RetinaScaleImageEffect extends ScaleImageEffect {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function applyEffect(ImageInterface $image) {
+    $target_with = $this->configuration['width'];
+    $target_height = $this->configuration['height'];
+    $upscale = $this->configuration['upscale'];
+    if (!$upscale) {
```

Comment:
You can inline these. It's very difficult to read. Had to move them to one line to understand what is going on...

Reply:
@imanoleguskiza You didn't do this.

Reply:
Done.